test/crd_validation_test.go

```diff
@@ -0,0 +1,77 @@
+package test
+
+import (
+	"github.com/RHsyseng/operator-utils/pkg/validation"
+	"github.com/ghodss/yaml"
+	qdrouterdv1alpha1 "github.com/interconnectedcloud/qdrouterd-operator/pkg/apis/interconnectedcloud/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+var crNameMap = map[string]string{
+	"interconnectedcloud_v1alpha1_qdrouterd_cr.yaml": "interconnectedcloud_v1alpha1_qdrouterd_crd.yaml",
+}
+
+var crdTypeMap = map[string]interface{}{
+	"interconnectedcloud_v1alpha1_qdrouterd_crd.yaml": &qdrouterdv1alpha1.Qdrouterd{},
+}
+
+func TestCRDSchemas(t *testing.T) {
+	for crdFileName, amqType := range crdTypeMap {
+		schema := getSchema(t, crdFileName)
+		missingEntries := schema.GetMissingEntries(amqType)
+		for _, missing := range missingEntries {
+			if strings.HasPrefix(missing.Path, "/status") {
```

Comment:
Not sure about this... since subresource / status is defined, why not declare the types in the CRD under status too?